### PR TITLE
ci: update `Stylelint` to v15

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-524]

- Updated stylelint from v13 to v15
  - https://stylelint.io/migration-guide/to-15
  - https://stylelint.io/migration-guide/to-14
- Added rules
  - [`keyframes-name-pattern: null`](https://stylelint.io/user-guide/rules/keyframes-name-pattern/)
    - Its null because we don't use kebab case. If we wanna restrict naming rule for this, we can do it
  - [`property-no-vendor-prefix: null`](https://stylelint.io/user-guide/rules/property-no-vendor-prefix/)
    - we use `-webkit-*` properties, so disabled the rule 
  - [`selector-class-pattern: null`](https://stylelint.io/user-guide/rules/selector-class-pattern/)
    - Disabled this rule because we use both kabab case for antd and camel case for our scss
  - [`selector-not-notation: 'simple'`](https://stylelint.io/user-guide/rules/selector-not-notation/)
    - Chose `simple` because we use `not` in this way
  - [`declaration-property-value-no-unknown: true`](https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown/)
    - Enabled this rule because i found a slew of invalid css usage. This rule sisallows unknown values for properties within declarations.
- Removed rules (all of the following rules are deprecated)
  - `at-rule-semicolon-space-before`
  - `declaration-block-semicolon-newline-after`
  - `declaration-block-semicolon-newline-before`
  - `declaration-block-semicolon-space-before`
  - `declaration-block-trailing-semicolon`
  - `no-eol-whitespace`
  - `no-extra-semicolons`
  - `string-quotes`

related: #5238: this is reverted due to CRACO. CRACO restricted library version for `postcss`

## Test Plan
- Check if stylelint works on local and ci
- Check if all migration works correctly
  - no UI regression

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-524]: https://determinedai.atlassian.net/browse/WEB-524?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/.stylelintrc.js [hunk 1]
```diff
@@ -1,2 +1,34 @@
-const config = require('./src/shared/configs/.stylelintrc');
-module.exports = config;
+module.exports = {
```
Comment (keita-determined) [inline webui/react/.stylelintrc.js:1]: stop importing from `shared` config since those are incompatible

File: webui/react/package.json [hunk 2]
```diff
@@ -122,15 +122,16 @@
     "magic-string": "^0.27.0",
     "morgan": "^1.10.0",
     "npm-force-resolutions": "0.0.10",
+    "postcss": "^8.4.21",
```
Comment (gt2345) [inline webui/react/package.json:125]: Where do we use this?

Comment (keita-determined) [reply webui/react/package.json]: stylelint v14 required `postcss` with `TypeError: Class extends value undefined is not a constructor or null` error. However, it seems like v15 doesnt require it. uninstalled it. nice finding

**Review by gt2345 — APPROVED** [review]
Great work. No error when building locally and no obvious regression.

**Pull request merged**